src/entry_table.rs

```diff
@@ -224,11 +226,10 @@ impl<const UID_LENGTH: usize, const KWI_LENGTH: usize> EntryTableValue<UID_LENGT
     >(
         &self,
         kwi_uid: &KmacKey,
-        max_uid_per_chain: usize,
     ) -> Vec<Uid<UID_LENGTH>> {
         let mut chain = Vec::new();
 
-        while chain.len() < max_uid_per_chain && chain.last() != self.chain_table_uid.as_ref() {
+        while chain.len() < MAX_UID_PER_CHAIN && chain.last() != self.chain_table_uid.as_ref() {
```

Comment:
Should we fail when reaching `MAX_UID_PER_CHAIN`? Here we'll silently return incomplete results, no?

Reply:
indeed, see 7950df0